src/ucs/arch/x86_64/cpu.c

```diff
@@ -588,6 +588,30 @@ static size_t ucs_cpu_memcpy_thresh(size_t user_val, size_t auto_val)
 }
 #endif
 
+#if ENABLE_NT_BUFFER_TRANSFER
```

Comment:
I've implemented it as a conditional to prevent the unnecessary use of two additional 64-bit variables when "NT_BUFFER_TRANSFER" is disabled. Please refer to the following definitions and structure declarations for more details
###########
#if ENABLE_NT_BUFFER_TRANSFER
#define UCS_ARCH_GLOBAL_OPTS_INITALIZER { \
    .builtin_memcpy_min     = UCS_MEMUNITS_AUTO, \
    .builtin_memcpy_max     = UCS_MEMUNITS_AUTO, \
    .nt_buffer_transfer_min = UCS_MEMUNITS_AUTO, \
    .nt_dest_threshold      = UCS_MEMUNITS_AUTO  \
}
#else
#define UCS_ARCH_GLOBAL_OPTS_INITALIZER { \
    .builtin_memcpy_min     = UCS_MEMUNITS_AUTO, \
    .builtin_memcpy_max     = UCS_MEMUNITS_AUTO  \
}
#endif

/* built-in memcpy & nt-buffer-transfer config */
typedef struct ucs_arch_global_opts {
    size_t builtin_memcpy_min;
    size_t builtin_memcpy_max;
#if ENABLE_NT_BUFFER_TRANSFER
    size_t nt_buffer_transfer_min;
    size_t nt_dest_threshold;
#endif
} ucs_arch_global_opts_t;
###########
If you believe this is excessive, I can modify it so that these are always compiled. What are your thoughts on this?

The ERMS memcpy instructions behave differently on various architectures and also depend on the length of the transfer. We should have named it ERMS_MEMCPY instead of BUILTIN_MEMCPY :). IMO these two methods should coexist, and the user should have the flexibility to use both simultaneously depending on the transfer length.(This would require platform-specific tuning to determine which method performs best for a given range)

Reply:
i think adding these variables is not an issue, and better make the code simpler and reduce the scope of preprocessor ifdef to the minimum

Reply:
Ok. Thanks for the clarification, I will remove this.